pkg/controller/vitesscell/reconcile_vtgate.go

```diff
@@ -116,6 +116,12 @@ func (r *ReconcileVitessCell) reconcileVtgate(ctx context.Context, vtc *planetsc
 	}
 	update.Annotations(&annotations, vtc.Spec.Gateway.Annotations)
 
+	// Merge ExtraVitessFlags into ExtraFlags.
```

Comment:
It seems that the same set of `ExtraVitessFlags` are intended to be passed to vtctld, vtgate and vttablet. That won't work because each component accepts a distinct (though not disjoint) set of flags. Passing a flag that a component doesn't accept will result in a failure to execute the binary.

Reply:
This feature is intended for passing flags that are shared by all components - such as topo flags (that's the specific use case we are trying to support in the operator with this feature).  I'll improve the comments to be more explicit about how `ExtraVitessFlags` is intended only for flags that can be passed to `vtctld`, `vtgate`, and `vttablet`.

Reply:
Added more clarification in the doc comment on the root `ExtraVitessFlags` https://github.com/planetscale/vitess-operator/pull/14/commits/f966fb07056e669256ffad20ec674fb2d83e881f.